File: lei_lds2122_grupo2-main/Frontend/src/app/models/FavoriteRoute.ts
/**
 * This model represents an user's favorite route 
 */
export class FavoriteRoute {

    /**
     * Construct's an instance of {@link FavoriteRoute}
     * 
     * @param userName the user that has the route marked as favorite
     * @param routeId the route marked as favorite
     */
    constructor(
        public userName: string,
        public routeId: string
    ) { }
}

File: lei_lds2122_grupo2-main/Backend/Data/Models/FavoriteRoute.cs

namespace Backend.Data.Models
{
    public class FavoriteRoute
    {
        public FavoriteRoute(string userName, long routeId)
        {
            UserName = userName;
            User = null!;
            RouteId = routeId;
            Route = null!;
        }


        //Navigations Properties 


        public string UserName { get; set; }//foreignKey
        public virtual User User { get; set; }//foreignKey

        public long RouteId { get; set; }//foreignKey   
        public virtual Route Route { get; set; }//foreignKey           

    }
}